js/component-courses.js

```diff
@@ -1,2 +1,36 @@
 // export as global since we're not using a module system
-window.Courses = { template: '<div>Classes</div>' }
+window.Courses = Vue.component("Courses", {
+    name: "Courses",
+    data: function() {
+        return {
+            list: null,
+            err: null
+        };
+    },
+    /* Before this view is rendered, fetch the data from the php file */
+    beforeRouteEnter(to, from, next) {
+        /* Fetch the returned json object value. */
+        fetch("get_info.php")
+            .then(function(response) {
+                response.json().then(function(data) {
+                    next(function(vm) {
+                        vm.list = data.DATA;
```

Comment:
Check the type of `data` before assigning it to a list. Handle the wrong type by throwing the appropriate error.